### DISCUSSION TITLE
Is it possible to set a "binary" header?

### DISCUSSION THREAD

**Discussion by Solid-Water-Dynamic** [ktr0731/evans#497] [Q&A] [answered]
Hello, it seems like the value of a header with its name ending with `-bin` is _Base64_ encoded when it's sent over the network.
For such a binary header, is it possible to set the value without _Evans_ doing _Base64_ encoding?
Or, is there a way to specify binary header's value? I tried \x or \u but they do not seem to work.

Thanks!

**Comment by ktr0731** [comment, marked as answer]
Hi. Unfortunately, we can't send binary header values directly. Its behavior is based on the gRPC spec.
According to [PROTOCOL-HTTP2.md](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md), it says:

> Note that HTTP2 does not allow arbitrary octet sequences for header values so binary header values must be encoded using Base64 as per https://tools.ietf.org/html/rfc4648#section-4. Implementations MUST accept padded and un-padded values and should emit un-padded values. Applications define binary headers by having their names end with "-bin". Runtime libraries use this suffix to detect binary headers and properly apply base64 encoding & decoding as headers are sent and received.

So, we can neither send binary header values directly nor `-bin`-prefixed headers without Base64 encoding.

**Reply by Solid-Water-Dynamic** [reply]
Thanks for the prompt answer, @ktr0731
